### ISSUE TITLE
Use pytest for tests?

### ISSUE THREAD

**Issue by blueyed** [agiliq/Django-parsley#50]
The current runtests.py script does not work for Django 1.7 (or 1.8?) anymore, and while at it I've wondered what's your opinion on using pytest for tests?

**Comment by mauricioabreu** [comment]
I don't know what is his opinion but I would put my efforts to make it happen.

**Comment by blueyed** [comment]
btw: there will be a "Adopt pytest month" probably:
https://mail.python.org/pipermail/pytest-dev/2015-February/002763.html

**Comment by mauricioabreu** [comment]
Yeah, it is a good time to do it but I don't if it is safe to contribute to this project. Author is missing so he is not even giving any feedback on pull requests and issues.

**Comment by blueyed** [comment]
@mauricioabreu 
We could always create a (temporary) fork with our improvements / fixes.

But you're right about the unresponsiveness from the @agiliq team.

**Comment by akshar-raaj** [comment]
@blueyed @mauricioabreu Please accept my apologies about our unresponsiveness. Looking into your older comments, please feel free to send a PR. We plan to maintain this app so your PR will not go unnoticed.

**Comment by blueyed** [comment]
@akshar-raaj 
This is good to hear. Thanks for letting us know.

**Comment by mauricioabreu** [comment]
hey @akshar-raaj of course I accept your apologies. :)
That is not your fault or anyone's else. We are just trying to make this app better and better.

I am currently migrating runtests.py to pytest (conftest). Actually only one test is failing using Django 1.7.x
After figuring out the problem I will change how tests are written. Should I do it step by step?

**Comment by blueyed** [comment]
@mauricioabreu 
Great. I think it would be great to push a separate branch for the pytest integration.

I would say that for the beginning existing tests should not get rewritten, but should work with the Django testrunner and pytest. But then new tests might be written using pytest features, if it gets accepted/merged.

**Comment by blueyed** [comment]
Re Dajngo 1.7:
The failing test for Django 1.7 is "test_radio_select (parsley.tests.tests.TestRadioSelect)".

Please see https://github.com/agiliq/Django-parsley/pull/49 for a WIP branch to get Django 1.7 support.

**Comment by mauricioabreu** [comment]
@blueyed Agreed.
I think your pull request could be merged. Django 1.7 is a important thing.

**Comment by mauricioabreu** [comment]
Hey @blueyed  now I see what you said.
Yes, I had the same error. How can we proceed?  Get your PR merged, so we can at least have the test suite using Django 1.7 and after that create a branch pytest (like I did but did not push yet) and keep coding there?

**Comment by blueyed** [comment]
@mauricioabreu 
The patch to `runtests.py` is quite small, and the rest from the django-1.7 branch (currently) is for travis only.

You could create the pytest-branch from master, and then cherry-pick just https://github.com/blueyed/Django-parsley/commit/e1c1b5ad71c8b6c9d68f4c74b6a5fd8f42994da0.
Or branch it off the PR's branch, but that might get rebased later and not merged as-is.
But then it would still be possible to get commits from the pytest branch into master / a PR.

But yes, merging the PR #49 already would simplify things.

@akshar-raaj 
Can this be done in the next days?

**Comment by mauricioabreu** [comment]
I got it @blueyed 
Let's wait for the owners decision.

**Comment by blueyed** [comment]
@akshar-raaj 
ping?

**Comment by mauricioabreu** [comment]
:(

**Comment by akshar-raaj** [comment]
@shivakrshn49 I see you have merged this PR. Is the issue done and can we close it?

**Comment by jproffitt** [comment]
This issue seems to be resolved, yes?

**Comment by blueyed** [comment]
not according to https://github.com/agiliq/Django-parsley/search?utf8=%E2%9C%93&q=pytest&type=
